skops/io/_utils.py

```diff
@@ -156,15 +155,19 @@ def get_state(value, save_context):
     # This is a helper function to try to get the state of an object. If it
     # fails with `get_state`, we try with json.dumps, if that fails, we raise
     # the original error alongside the json error.
+
+    # TODO: This should help with fixing recursive references.
+    # if id(value) in save_context.memo:
+    #     return {
+    #         "__module__": None,
+    #         "__class__": None,
+    #         "__id__": id(value),
+    #         "__loader__": "CachedNode",
+    #     }
```

Comment:
I think this should work ok, but we would need to restructure how things get memoized.

Right now, things only get memoized once they've been fully constructued, so if an object has a recursive attribute, it never actually gets to be in the memo.

I thought about this originally, and started exploring if there's a nice way to hold a reference to an object that isn't initialized yet, so we could do something like:


```
    _obj = Object()  # pseudocode, not the right way to do this

    save_context.memoize[_obj, id]
    res = _get_state(value, save_context)

    _obj.assign(res)   

    return _obj
```

Still haven't fully thought this through, but I think we would need to do something like this to allow circular self-references to work the right way.